daemon/src/collab_settlement_maker.rs

```diff
@@ -192,6 +199,8 @@ impl Actor {
             .await
             .context("Failed to inform taker about settlement acceptance")??;
 
+        self.has_accepted = true;
```

Comment:
Somehow I would feel more comfortable if we would do this right after the check.